### PR TITLE
Raise error when invalid/unsupported units supplied #157

### PR DESCRIPTION
**Pull request by kikihakiem**
Use Geokit::default_units only if no units supplied.

### CODE REVIEW

**Comment by kikihakiem** [comment]
@houndci is it better to meld the fixes into previous commit, or just push another commit?

File: lib/geokit/mappable.rb [hunk 1]
```diff
@@ -86,7 +91,11 @@ def heading_between(from, to)
       # an endpoint. Returns a LatLng instance. Typically, the instance method
       # will be used instead of this method.
       def endpoint(start, heading, distance, options = {})
-        units   = options[:units] || Geokit.default_units
+        units = options[:units]
+        units = Geokit::default_units if units.nil?
+        [:miles, :kms, :meters, :nms].include?(units) or raise ArgumentError.new(
```
Comment (kikihakiem) [inline lib/geokit/mappable.rb]: Don't you think that `or` in this context is a control flow?

**Comment by mnoack** [comment]
@kikihakiem Thanks very much for the PR. I will merge it regardless, but I would really love if you could extract the duplicate checking of units into a method, e.g. units = get_units!(options[:units])

I think you can also just say:
raise ArgumentError, "error message here" instead of using ".new("

**Comment by mnoack** [comment]
It's up to you if you follow houndci or not, I find apart from lines-too-long the suggestions are usually good, but use your judgement after it's comments.

**Comment by kikihakiem** [comment]
Sure, I will extract it to a helper method. Thanks for your review.

**Comment by kikihakiem** [comment]
Done. Silly me, replying to houndci's comments. Just find out that it's a bot after visiting houndci.com lol

**Comment by kikihakiem** [comment]
@mnoack could you please review this?

**Comment by mnoack** [comment]
Thanks @kikihakiem - I generally don't do gem releases unless a bit has built up or it's needed by people so let me know. You can always use gems via bundle via the "git" option.

**Pull request merged**